lib/evaluateFunctions.js

```diff
@@ -65,6 +68,229 @@ function evaluateFunctions(node) {
+      const exponentNode = radicandNode.args[1];
+      // TODO(ael): this only works if they actually have a value...
+      // ideally I'd use equals, but the nodes are different for some reason
+      // even if the value is the same...
+      if (rootNode.value === exponentNode.value) {
+        newNode = baseNode;
+        return NodeStatus.nodeChanged(
+          MathChangeTypes.NTH_ROOT_VALUE, oldNode, newNode);
+      } else {
+        return NodeStatus.noChange(node);
+      }
+    }
+    else {
+      throw Error('Operation not supported: ' + radicandNode.op);
+    }
+  }
+  else if (NodeType.isConstant(radicandNode)) {
+    // if it's negative, don't do anything (until imaginary number support)
+    if (Negative.isNegative(radicandNode)) {
+      return NodeStatus.noChange(node);
+    }
+    else if (!NodeType.isConstant(rootNode) || Negative.isNegative(rootNode)) {
+      return NodeStatus.noChange(node);
+    }
+
+    const radicandValue = parseFloat(radicandNode.value);
+    const rootValue = parseFloat(rootNode.value);
+    const nthRootValue = math.nthRoot(radicandValue, rootValue);
+    if (nthRootValue % 1 === 0) {
+      newNode = NodeCreator.constant(nthRootValue);
+      return NodeStatus.nodeChanged(
+        MathChangeTypes.NTH_ROOT_VALUE, oldNode, newNode);
+    }
+    else {
+      const subSteps = [];
+
+      // 1. convert the number into the product of its prime factors
+      const factors = Factor.getPrimeFactors(radicandValue);
+      const factorNodes = factors.map(factor => {
+        return NodeCreator.constant(factor);
+      })
+
+      newNode.args[0] = NodeCreator.operator('*', factorNodes);
+      subSteps.push(NodeStatus.nodeChanged(
+          MathChangeTypes.FACTOR_INTO_PRIMES, oldNode, newNode));
+
+      // TODO: Consider moving steps 2, 3 and 4 into distributing the
+      // radical into children above, and just calling nthNode on the
+      // factored version
+
+      // 2. group any prime factors that show up root times
+      oldNode = NodeStatus.resetChangeGroups(newNode);
+      newNode = clone(oldNode, false);
+
+      let groupedFactors = [];
+      let hasGroups = false;
+      let i = 0;
+      while (i < factors.length) {
+        let j = i;
+        while (j < factors.length) {
+          if (factors[i] !== factors[j]) {
+            break;
+          }
+          j++;
+        }
+        if (j - i >= rootValue) {
+          let groupEnd = i + rootValue;
+          groupedFactors.push(factors.slice(i, groupEnd));
+          hasGroups = true;
+          i = groupEnd;
+        }
+        else {
+          groupedFactors = groupedFactors.concat(factors.slice(i, j));
+          i = j;
+        }
+      }
+
+      if (!hasGroups) {
+        return NodeStatus.noChange(node);
+      }
+
+      let children = [];
+      for (let i = 0; i < groupedFactors.length; i++) {
+        const item = groupedFactors[i];
+        if (groupedFactors[i].constructor === Array) {
```

Comment:
ohh I see later why you do this
might not be obvious to the student at this stage. pedagogy details though, can always change later